### PR TITLE
#41 Create API

### LINKED ISSUE #41: Build API
### Description

Build Flask API to allow technical users to query the earthquake data in real-time for use in other applications.
- Use the API README document for guidance on endpoints 

---

### Outcome 

A Flask API that can be queried in real time 

---

### User Story
As a technical user, I want to be able to query a dataset with a variety of query parameters to carry out data exploration and extract the information I want.

### PR DESCRIPTION
**Pull request by Fahi28**
Initial endpoints for the API from the README (excluding region since we have decided to not implement that relationship with earthquakes). Also created a database.py file to query the database for the API.

Closes #41

### CODE REVIEW

File: api/api.py [hunk 1]
```diff
@@ -1 +1,104 @@
...
+    sort = request.args.get("sort")
+
+    if not start_date:
+        return jsonify({"error": "start_date must be given"}), 404
+
+    try:
+        start_date = datetime.strptime(
+            start_date, '%Y-%m-%d').strftime('%Y-%m-%d')
+    except ValueError:
+        return jsonify({"error": "Invalid start_date format. Please use MM/DD/YYYY"}), 400
```
Comment (Jakub-Poskrop) [inline api/api.py:66]: Shouldn't this say `please use YYYY-MM-DD` or the strptime format be `%m/%d/%Y`?

File: api/api.py [hunk 2]
```diff
@@ -1 +1,104 @@
...
+
+    if not end_date:
+        return jsonify({"error": "end_date must be given"}), 404
+
+    try:
+        end_date = datetime.strptime(
+            end_date, '%Y-%m-%d').strftime('%Y-%m-%d')
```
Comment (Jakub-Poskrop) [inline api/api.py:73]: Same here?

**Review by Jakub-Poskrop — APPROVED** [review]
API looks good

File: api/api.py [hunk 3]
```diff
@@ -1 +1,104 @@
...
+    earthquakes = get_earthquakes_by_magnitude(min_magnitude, max_magnitude)
+
+    if not earthquakes:
+        return jsonify({"error": "No earthquakes found"}), 404
+
+    return jsonify(earthquakes), 200
+
+
+@app.route("/earthquakes/date", methods=["GET"])
+def endpoint_get_earthquakes_between_date():
```
Comment (S1mpySloth) [inline api/api.py:54]: Docstring perhaps?

File: api/api.py [hunk 4]
```diff
@@ -1 +1,104 @@
...
+
+    earthquakes = get_earthquakes_by_date(start_date, end_date, sort)
+
+    if not earthquakes:
+        return jsonify({"error": "No earthquakes found"}), 404
+    return jsonify(earthquakes), 200
+
+
+@app.route("/earthquakes/alert/<string:color>")
+def endpoint_earthquakes_by_alert_color(color: str):
```
Comment (S1mpySloth) [inline api/api.py:90]: Docstring for extra documentation?

File: api/api.py [hunk 5]
```diff
@@ -1 +1,104 @@
...
+    earthquake = get_earthquake_by_id(earthquake_id)
+
+    if not earthquake:
+        return jsonify({"error": "Earthquake not found"}), 404
+
+    return jsonify(earthquake), 200
+
+
+@app.route("/earthquakes/magnitude", methods=["GET"])
+def endpoint_get_magnitude():
```
Comment (Jakub-Poskrop) [inline api/api.py:35]: Possibly add error handling if someone tries making `min_magnitude` larger than `max_magnitude`?

Comment (Jakub-Poskrop) [reply api/api.py:35]: Also are there any boundaries to what value they can give?

**Review by S1mpySloth — APPROVED** [review]
Looks good to me! Docstrings added where suggested :D

**Review by Jakub-Poskrop — APPROVED** [review]
Looks good

**Pull request merged**